ReactWindows/ReactNative/Chakra/Executor/NativeJavaScriptExecutor.cs

```diff
@@ -105,14 +133,57 @@ public void RunScript(string sourcePath, string sourceUrl)
 
             try
             {
-                if (_useSerialization)
-                {
-                    var binPath = Path.Combine(ApplicationData.Current.LocalFolder.Path, "ReactNativeBundle.bin");
-                    Native.ThrowIfError((JavaScriptErrorCode)_executor.RunSerializedScript(sourcePath, binPath, sourceUrl));
-                }
-                else
+                var binPath = Path.Combine(ApplicationData.Current.LocalFolder.Path, BytecodeFileName);
+
+                switch (_serializationMode)
                 {
-                    Native.ThrowIfError((JavaScriptErrorCode)_executor.RunScript(sourcePath, sourceUrl));
+                    case SerializationMode.Foreground:
+                        Native.ThrowIfError((JavaScriptErrorCode)_executor.RunSerializedScript(sourcePath, binPath, sourceUrl));
+                        break;
+
+                    case SerializationMode.Background:
+                        var binFileInfo = new FileInfo(binPath);
+
+                        if (binFileInfo.Exists)
+                        {
+                            // This function checks if the bytecode file matches the JS bundle and makes
+                            // a new bytecode file if necessary, but it does this on the same thread.
+                            Native.ThrowIfError((JavaScriptErrorCode)_executor.RunSerializedScript(sourcePath, binPath, sourceUrl));
```

Comment:
I don't think this is desired behavior. This could result in a new bundle being created synchronously, e.g., when using CodePush. I think we probably need two separate APIs, SerializeScript and RunSerializedScript, the latter will just fail if the binary file does not exist. You'll need to do all the file creation time checking in the managed proxy.

Reply:
Agreed.